runtime/src/commonMain/kotlin/com/squareup/sqldelight/db/SqlDriver.kt

```diff
@@ -90,3 +92,35 @@ interface SqlDriver : Closeable {
     fun migrate(driver: SqlDriver, oldVersion: Int, newVersion: Int)
   }
 }
+
+class AfterVersion(
+  val afterVersion: Int,
+  val block: () -> Unit
+)
+
+fun SqlDriver.Schema.migrateWithCallbacks(
+  driver: SqlDriver,
+  oldVersion: Int,
+  newVersion: Int,
+  vararg callbacks: AfterVersion
+) {
+  var lastVersion = oldVersion
+  callbacks.sortedBy { it.afterVersion }.forEach { callback ->
+    migrate(
+      driver = driver,
+      oldVersion = max(lastVersion, oldVersion),
+      newVersion = min(callback.afterVersion + 1, newVersion),
+    )
+
+    if (callback.afterVersion in oldVersion until newVersion) {
+      callback.block()
+    }
+
+    lastVersion = callback.afterVersion + 1
+  }
+  migrate(
+    driver = driver,
+    oldVersion = max(lastVersion, oldVersion),
+    newVersion = newVersion
+  )
+}
```

Comment:
The logic of this function is very hard to follow with the max and mins as well as the conditional callback. What do you think about this instead:
```kotlin
var lastVersion = oldVersion

// For each callback within the oldVersion..newVersion range, alternate between migrating
// the schema and invoking each callback.
callbacks.filter { it.afterVersion > oldVersion && it.afterVersion <= newVersion }
  .sortedBy { it.afterVersion }
  .forEach { callback ->
    migrate(driver, old = lastVersion, new = it.afterVersion)
    callback.block()
    lastVersion = it.afterVersion
  }

// If there were no callbacks, or the newVersion is higher than the highest callback,
// complete the migration.
if (lastVersion < newVersion) {
  migrate(driver, lastVersion, newVersion)
}
```